clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -552,8 +555,63 @@ public void testDelayedReconciliationResultAppliedWhenTargetChangedWithMetadataU
 
         // Pending assignment that was discovered in metadata should be ready to reconcile in the
         // next reconciliation loop.
-        Set<TopicIdPartition> topic2Assignment = topicIdPartitionsSet(topicId2, topic2, 1, 2);
-        assertEquals(topic2Assignment, membershipManager.assignmentReadyToReconcile());
+        Map<Uuid, SortedSet<Integer>> topic2Assignment = topicIdPartitionsMap(topicId2,  1, 2);
+        assertEquals(topic2Assignment, membershipManager.topicPartitionsAwaitingReconciliation());
+    }
+
+    // Tests the case where topic metadata is not available at the time of the assignment,
+    // but is made available later.
+    @Test
+    public void testDelayedMetadataUsedToCompleteAssignment() {
```

Comment:
I wonder if having the same test but starting with an empty assignment would bring any value. Have you considered it? For the context, I was wondering if the second reconciliation would be started if the metadata for the second topic is received while the first topic is reconciled.

Reply:
If I understand your ask correctly, I think the test wouldn't pass yet. If the metadata arrives while the first topic is being reconciled, we'll not retrigger reconciliation, unless we get another assignment via the heartbeat. I will fix this and implement the test you are suggesting in [KAFKA-15832](https://issues.apache.org/jira/browse/KAFKA-15832)), where we'll check for the need to reconcile regularly during poll.